### PR TITLE
compilation: automate applying compiler bug workarounds

### PR DESCRIPTION
**Pull request by wence-**
Rather than hard-coding for everyone, select correct "bugfix" flags
depending on the compiler version.

### CODE REVIEW

**Comment by miklos1** [comment]
More "workaround" than "bugfix".

File: pyop2/compilation.py [hunk 1]
```diff
@@ -79,10 +117,31 @@ def __init__(self, cc, ld=None, cppargs=[], ldargs=[],
...
+    def bugfix_cflags(self):
+        """Flags to work around bugs in compilers."""
+        compiler, ver = self.compiler_version
+        if compiler == "gcc":
+            if version.StrictVersion("4.8.0") <= ver < version.StrictVersion("4.9.0"):
+                # GCC bug https://gcc.gnu.org/bugzilla/show_bug.cgi?id=61068
+                return ["-fno-ivopts"]
+            if version.StrictVersion("6.0.0") <= ver < version.StrictVersion("7.0.1"):
+                # GCC bug https://gcc.gnu.org/bugzilla/show_bug.cgi?id=79920
+                return ["-fno-tree-loop-vectorize"]
```
Comment (dorugeber) [inline pyop2/compilation.py]: `-fno-tree-loop-vectorize` or `-fno-tree-vectorize`?

Comment (wence-) [inline pyop2/compilation.py]: The former appeared to work for me?

Comment (dorugeber) [inline pyop2/compilation.py]: Okay, the GCC guy on the bug report mentioned the latter, that's all.

**Pull request merged**